generate-stackbrew-library.sh

```diff
@@ -10,33 +10,77 @@ cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
 
 versions=( */ )
 versions=( "${versions[@]%/}" )
-url='git://github.com/docker-library/golang'
 
-echo '# maintainer: InfoSiftr <github@infosiftr.com> (@infosiftr)'
-echo '# maintainer: Johan Euphrosine <proppy@google.com> (@proppy)'
+# get the most recent commit which modified any of "$@"
+fileCommit() {
+	git log -1 --format='format:%H' HEAD -- "$@"
+}
+
+# get the most recent commit which modified any of "$1/Dockerfile" or any file COPY'd from "$1/Dockerfile"
+dirCommit() {
+	local dir="$1"; shift
+	(
+		cd "$dir"
+		fileCommit \
+			Dockerfile \
+			$(git show "$ref":./Dockerfile | awk '
+				toupper($1) == "COPY" {
+					for (i = 2; i < NF; i++) {
+						print $i
+					}
+				}
+			')
+	)
+}
+
+self="$(basename "$BASH_SOURCE")"
+cat <<-EOH
+# this file is generated via https://github.com/docker-library/golang/blob/$(fileCommit "$self")/$self
```

Comment:
Yeah, I've discussed this a lot with @paultag -- the format we're using is technically only loosely based on RFC 2822 -- it's been called "deb822" or "deb2822" before as well, since it's so common in Debian, but we're still trying to think up a good name that @paultag is happy with. :smile: